File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/WildEgg.java
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet;

/* JADX INFO: loaded from: classes3.dex */
public class WildEgg extends it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Consumable, it.paranoidsquirrels.idleguildmaster.storage.data.items.Item
    protected void configureProperties() {
        this.idName = R.string.egg_wild_name;
        this.idDescription = R.string.egg_wild_description;
        this.idImage = R.drawable.egg_wild;
        this.source.add(Integer.valueOf(R.string.dungeon_name_enchanted_forest));
        this.source.add(Integer.valueOf(R.string.dungeon_name_blackwater_port));
        this.price = 100L;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
    public Pet hatch() {
        int iCalculateNewPetId = Utils.calculateNewPetId();
        double dRandom = Utils.random();
        if (dRandom < 0.75d) {
            return Pet.getInstance("Rat", iCalculateNewPetId);
        }
        return dRandom < 0.95d ? Pet.getInstance("Squirrel", iCalculateNewPetId) : Pet.getInstance("RedWolf", iCalculateNewPetId);
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/items/instances/WildEgg.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.items.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Egg
import it.paranoidsquirrels.idleguildmaster.storage.data.pets.Pet

class WildEgg : Egg() {
    override fun configureProperties() {
        idName = R.string.egg_wild_name
        idDescription = R.string.egg_wild_description
        idImage = R.drawable.egg_wild
        source.add(R.string.dungeon_name_enchanted_forest)
        source.add(R.string.dungeon_name_blackwater_port)
        price = 100L
    }

    override fun hatch(): Pet? {
        val id = Utils.calculateNewPetId()
        val r = Utils.random()
        return if (r < 0.75) {
            Pet.getInstance("Rat", id)
        } else if (r < 0.95) {
            Pet.getInstance("Squirrel", id)
        } else {
            Pet.getInstance("RedWolf", id)
        }
    }
}
